src/modules/NtpClient.rb

```diff
@@ -1099,10 +1099,12 @@ def update_cfa_record(record)
     def write_ntp_conf
       records_for_write.each do |record|
         unless record["cfa_record"]
-          record["cfa_record"] = CFA::NtpConf::Record.record_class(record["type"]).new
-          ntp_conf.records << record["cfa_record"]
+          ntp_conf.records << CFA::NtpConf::Record.record_class(record["type"]).new
+          record["cfa_record"] = ntp_conf.records.last
```

Comment:
Yet you felt brave enough to change CFA incompatibly, and [it will happen again for 1.0](https://github.com/config-files-api/config_files_api/issues/17). There was a bug not covered with tests that you had to find by manually running the module. Come on, write the test!